runtime/compiler/ilgen/Walker.cpp

```diff
@@ -5380,6 +5380,27 @@ break
          }
          _intrinsicErrorHandling = false;
       }
+
+   // The call may be transformed into a non-OSR point. Check if bookkeeping is needed
+   // before the transformation.
+   bool needOSRBookkeeping = false;
+   int32_t osrInductionOffset;
+
+   // callNodeTreeTop may be null if this call should not be placed in the trees
+   if (callNodeTreeTop
+       && comp()->getOption(TR_EnableOSR)
+       && !comp()->isPeekingMethod()
+       && comp()->isOSRTransitionTarget(TR::postExecutionOSR)
+       && comp()->isPotentialOSRPoint(callNodeTreeTop->getNode())
+       && !_methodSymbol->cannotAttemptOSRAt(callNode->getByteCodeInfo(), NULL, comp())
+       && !_cannotAttemptOSR)
+      {
+      needOSRBookkeeping = true;
+      // callNode may become a non-OSR point after the transformation, calling getOSRInductionOffset
+      // on a non-OSR point will trigger the assertion, thus get the induction offset here.
+      osrInductionOffset = comp()->getOSRInductionOffset(callNode);
```

Comment:
Can you please share a case where the call node becomes something other than a call and we need the OSR point - ie we will still have a point where we want to decompile and that point is at an invalidation point? If it isn't an invalidation point why do we want the transition point?

Reply:
`J9VMInternals.getSuperclass` will be transformed to a different call. A call to `JITHelpers.jitHelpers()` is also inserted. In this case, the invalidation point still exists but the calls are not OSR points. Even if the invalidation point is removed, there is no harm in doing the OSR bookkeeping and inserting the OSR helper call. Any redundant helper calls will be removed before inserting OSR guards.

Reply:
right it just adds overhead, but given the perf numbers I guess we can live with that